openadapt/ml/huggingface_models/generic_provider.py

```diff
@@ -0,0 +1,49 @@
+"""Generic CompletionProvider module."""
+
+from transformers import pipeline
+
+from openadapt.ml.base_provider import (
+    Availability,
+    Capability,
+    CompletionProvider,
+    Modality,
+)
+
+
+class GenericHuggingFaceProvider(CompletionProvider):
+    """GenericHuggingFaceProvider class."""
+
+    Name: str = "Generic HuggingFace Provider"
+    Capabilities: list[Capability] = [
+        Capability.TRAINING,
+        Capability.TUNING,
+        Capability.INFERENCE,
+    ]
+    Modalities: list[Modality] = [Modality.TEXT]
+    Availabilities: list[Availability] = [Availability.HOSTED]
```

Comment:
Should this be `Availabilities: list[Availability] = [Availability.LOCAL_CPU, Availability.LOCAL_GPU]`?

We would need a separate provider for HOSTED models.

Can you please rename this `LocalHuggingFaceProvider`, and add a TODO above the class to implement one or more `HostedHuggingFaceProvider` to support Inference API and Inference Endpoints at https://huggingface.co/docs/huggingface_hub/guides/inference?